R/create_task_makefile.R

```diff
@@ -127,6 +112,46 @@ create_task_makefile <- function(
       tasks[[task]]$steps[[step]]$has_depends <- length(tasks[[task]]$steps[[step]]$depends) > 0
     }
   }
+  pos_extensions <- unique(c(remake::file_extensions(), file_extensions))
+  
+  formatted_final_targets <- sapply(X = targets_to_combine, function(x){
+    if (remake:::target_is_file(x, file_extensions = pos_extensions)){
+      paste0("'", x, "'")
+    } else {
+      x
+    }
+  })
+  
+  if (is.null(finalize_fun)){
+    # when there is no finalize fun, the "jobs" are the final step targs and they have no new commands
+    job <- lapply(seq_len(length(formatted_final_targets)), function(i){
+      list(
+        target_name = unname(formatted_final_targets[i])
+      )
+      })
+    has_finalize_funs <- FALSE
+  } else {
+    has_finalize_funs <- TRUE
+    job <- lapply(seq_len(length(finalize_fun)), function(i){
+      list(
+        target_name = ifelse(as_promises, paste0(basename(target_name[i]), "_promise"), target_name[i]), 
+        command = {
+          to_combine <- paste(formatted_final_targets, collapse=',\n      ')
+          target_is_file <- remake:::target_is_file(target_name[i], file_extensions = pos_extensions)
+          # hide the object/file changes as input when using `promises`
+          if (target_is_file){
+            ifelse(as_promises, {
+              sprintf("%s(I('%s'),\n      %s)", finalize_fun[i], target_name[i], to_combine)
+            }, {
+              sprintf("%s(target_name,\n      %s)", finalize_fun[i], to_combine)
+            })
+          } else {
+            combine_str <- "%s(\n      %s)"
+            sprintf(combine_str, finalize_fun[i], to_combine) # use PR #90 when it is merged
```

Comment:
Could take this comment out now. For the record, I think if you really wanted to use psprintf, you could do:
```
do.call(
  scipiper::psprintf,
  c(
    list('%s('=finalize_fun),
    sprintf('%s,', to_combine[-length(to_combine)]),
    sprintf('%s)', to_combine[length(to_combine)])
  )
)
```
but I like your way better.